src/app-instance.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * app-timestamp.js
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// The way that the jest framework works, it has a pool of
+// processes which each one or more tests. This means that
+// there can be multiple process running at the same time.
+//
+// Furthermore, if one of the processes ends, then it may
+// start a new test. This means that the same process may
+// run multiple tests, one after the other.
+//
+// During tests, some portions of the gateway may require
+// a unique identifier for each test that is run, and that is
+// the purpose of this module.
+
+'use strict';
+
+const process = require('process');
+
+class AppInstance {
+
+  constructor() {
+    this.timestamp = this.mark();
```

Comment:
mark() doesn't return anything, so timestamp will be initialized to undefined. If this is fixed, the call to `mark()` in app.js can be removed

Reply:
Doh